src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -204,32 +251,475 @@ TEST_F(AlertRequestTest, Run_MandatoryParametersAreInvalid_UNSUCCESS) {
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      updateRequestTimeout(kConnectionKey, kCorrelationId, kDefaultTimeout));
+
+  MessageSharedPtr result_msg(
+      CatchMobileCommandResult(CallOnEvent(*command, event)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_ENUM,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, OnEvent_UIAlertHasHmiResponsesToWait_UNSUCCESS) {
+  AddAlertTextsToMsg();
+  AddTTSChunkToMsg();
+
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::WARNINGS;
+  (*msg_)[am::strings::msg_params][am::strings::play_tone] = true;
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(mock_message_helper_,
+              ProcessSoftButtons((*msg_)[am::strings::msg_params], _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(_))
+      .Times(AtLeast(1))
+      .WillRepeatedly(Return(true));
+
+  command->Run();
+
+  Event event(hmi_apis::FunctionID::UI_Alert);
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(mock_app_manager_,
+              ManageHMICommand(
+                  HMIResultCodeIs(hmi_apis::FunctionID::TTS_StopSpeaking)));
+
+  command->on_event(event);
+}
+
+TEST_F(AlertRequestTest, OnEvent_TTSWarnings_SUCCESS) {
+  AddTTSChunkToMsg();
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::WARNINGS;
+  (*msg_)[am::strings::msg_params][am::strings::play_tone] = true;
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(mock_message_helper_,
+              ProcessSoftButtons((*msg_)[am::strings::msg_params], _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(_))
+      .Times(AtLeast(1))
+      .WillRepeatedly(Return(true));
+
+  command->Run();
+
+  Event event(hmi_apis::FunctionID::TTS_Speak);
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::WARNINGS),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->on_event(event);
+}
+
+TEST_F(AlertRequestTest, OnEvent_TTSUnsupportedResource_SUCCESS) {
+  AddTTSChunkToMsg();
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg_)[am::strings::msg_params][am::strings::play_tone] = true;
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_FULL));
```

Comment:
@okozlovlux, `ON_CALL` for `hmi_level()` appears in almost every test case and take `HMI_FULL` value. May be it would be better to move it to constructor, or `SetUp()` method?